src/main/java/ether/TransactionsManager.java

```diff
@@ -0,0 +1,663 @@
+
+        if (Settings.DEBUG_INFO) {
+            System.out.println("Client-->SuperPeer: " + clientToSuperPeerChannel == null ? "null" : clientToSuperPeerChannel);
+        }
+
+        //Get client EtherBalance
+        String clientEtherBalance;
+        try {
+            clientEtherBalance = EtherClient.getEtherBalance(sourceId.toString(), httpAgent);
+        }catch (IOException e){
+            if (Settings.DEBUG_INFO) {
+                System.out.println("Failed to get Ether balance for: " + sourceId);
+            }
+            JSON.sendErrorResponse("Failed to get Ether balance.", sourceId, meshManager);
+            return;
+        }
+
+        //Get client Token balance
+        String clientTokenBalance;
+        try {
+            clientTokenBalance = EtherClient.getTokenBalance(sourceId.toString(), httpAgent);
+        }catch (IOException e){
+            if (Settings.DEBUG_INFO) {
+                System.out.println("Failed to get Token balance for: " + sourceId);
+            }
+            JSON.sendErrorResponse("Failed to get Token balance.", sourceId, meshManager);
+            return;
+        }
+
+        BigInteger clientNonce = EtherClient.getNonce(sourceId.toString(), httpAgent);
+        if(clientNonce == null){
+            if (Settings.DEBUG_INFO) {
+                System.out.println("Failed to get nonce for: " + sourceId);
+            }
+            JSON.sendErrorResponse("Failed to get nonce.", sourceId, meshManager);
+            return;
+        }
+
+        JSON.sendResponse(superPeerToClientChannel, clientToSuperPeerChannel, clientEtherBalance, clientTokenBalance, clientNonce, sourceId, meshManager);
+    }
+
+
+    /**
+     * Processes Open Client request from a Client-Remote Peer
+     * @param sourceId
+     * @param jsonObject
+     */
+    private void processOpenClientToSuperPeerRequest(MeshID sourceId, JSONObject jsonObject){
+
+        Object signedApproveTransaction = jsonObject.get("signedApproveTransaction");
+        if(signedApproveTransaction == null){
+            if (Settings.DEBUG_INFO) {
+                System.out.println("No signed approve transaction in the open channel request from client.");
+            }
+            JSON.sendErrorResponse("No signed approve transaction in the open channel request from client.", sourceId, meshManager);
+            return;
+        }
+
+
+        Object signedOpenChannelTransaction = jsonObject.get("signedOpenChannelTransaction");
+        if(signedOpenChannelTransaction == null){
+            if (Settings.DEBUG_INFO) {
+                System.out.println("No signed transaction in the open channel request from client.");
+            }
+            JSON.sendErrorResponse("No signed transaction in the open channel request from client.", sourceId, meshManager);
+            return;
+        }
+
+
+        //Calculate Client-->SuperPeer hash
+        byte[] clientToSuperPeerHash = EtherUtility.getChannelHash(sourceId.toString(), ownMeshId.toString());
```

Comment:
Is this duplicate code to line 351? If so these should probably be functions.

Reply:
Looks like a few other of these checks are duplicated in a few places as well (e.g. fetching `clientToSuperPeerChannel` from the blockchain) - moving the duplicate checks to their own functions will make this a lot more readable.